src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -40,23 +40,29 @@ public void init() {
   }
 
   /**
-   * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
-   * an empty array if the user is not provided.
+   * Responds with a JSON representation of {@link Message} data for logged-in user and another user. Responds with
+   * an empty array if the user is not logged in or another user is not provided.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     response.setContentType("application/json");
 
-    String user = request.getParameter("user");
-
-    if (user == null || user.equals("")) {
+    // Get logged-in user and other user
+    String loggedInUser = null;
+    String otherUser = request.getParameter("user");
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      loggedInUser = userService.getCurrentUser().toString();
```

Comment:
Can `getCurrentUser()` return null? If so, it looks like this code will throw `NullPointerException`.